lib/task_details_page.dart

```diff
@@ -41,6 +42,7 @@ class _TaskDetailsPageState extends State<TaskDetailsPage> {
     super.dispose();
   }
 
+  // Carlo
```

Comment:
You could extract date & time picker into a separate function.

You could use a reusable function for button styling.

Could use a FutureBuilder to show a CircularProgressIndicator while the task is being updated.

Could also wrap text fields in a form widget.

Could add a confirmation dialog for the different actions.